### DISCUSSION TITLE
Version 14.x is basically 13.x but will switch jakarta <-> javax ...

### DISCUSSION THREAD

**Discussion by rbygrave** [ebean-orm/ebean#3257] [General]
So `14.0.0` will be a **jakarta** artefact and `14.0.0-javax` will be a **javax** artefact.

We will still continue to release both a jakarta and javax version of ebean with each release.

I'm looking to do this because the master branch is jakarta now and I think this will be easier for me to get the releases correct and not make a mistake doing that plus makes it easier to put out some pre-release versions of 15.x (and I intend 15.x to be jakarta only at this stage).

With 14.x I'm also going to trial only releasing the ebean-agent, ebean-maven-plugin and ebean-gradle-plugin with minor versions (so 14.0, 14.1, 14.2) ... but still release the maven tile for every release.  Currently we release all these for every release even though they rarely change which is "simple" but also a bit "wasteful".  

**Summary, 14.x will:**
- switch Jakarta and Javax (so 14.0.0 is a Jakarta artefact and 14.0.0-javax is a Javax one)
- trial only releasing ebean-agent with minor versions (14.0, 14.1, 14.2 etc)
- change the default persist batch size from 20 to 200
- remove some deprecated methods maybe

**Comment by jnehlmeier** [comment]
> Currently we release all these for every release even though they rarely change which is "simple" but also a bit "wasteful".

But it is also easy for a developer do reason about it. Otherwise you need some sort of dependency matrix, showing which Ebean version works with which agent/plugin.

What if a plugin will be rewritten and introduces some breaking changes? Do we then have for example Ebean 14.x with plugin 15.x because of semantic versioning?

**Reply by rbygrave** [reply]
I believe (hence trial) that the vast majority access the ebean-agent via maven tile and gradle plugin.  So I should adjust the above to say we release those 2 on each release.

If it is too confusing then yeah we go back to also releasing the ebean-agent and ebean-maven-plugin on every release.

For 15.x there will be no actual difference in agent. When we will are releasing that we will be in the situation of releasing 3 sets of artifacts - 14.x (Jakarta) 14.x-javax, and 15.x (a cut down of the 14.x jakarta). 

> plugin will be rewritten and introduces some breaking changes? 

Any plugin change will apply to all 3 (14.x, 14.x-javax, and 15.x). There will not be a 15.x specific ebean-agent. (and ebean-agent stays backward compatible to 11.x and must stay using java 8).

15.x will stay aligned to 14.x with the difference being the features that have been removed. So bug fixes etc will be applied to 14.x and then 15.x.
